src/coreclr/src/vm/ceeload.cpp

```diff
@@ -1456,7 +1456,9 @@ BOOL Module::IsAlwaysSavedInPreferredZapModule(Instantiation classInst,     // t
     return ClassLoader::IsTypicalSharedInstantiation(classInst) &&
            ClassLoader::IsTypicalSharedInstantiation(methodInst);
 }
+#endif // FEATURE_PREJIT
 
+#if defined(FEATURE_PREJIT) || defined(FEATURE_MULTICOREJIT_GENERIC)
```

Comment:
It does not sound right to make MultiCoreJit depend on Preffered Zap Module concept.

Reply:
I checked LoaderModule, but it is not proper. I will find another way such as module of caller and etc.
And if you have any good idea, please let me know. Thank you.

Reply:
Can the multi-core JIT encoding start with the method's Module (ie `MethodDesc::GetModule()`)?

Reply:
I cannot find other way yet. I updated to MethodDesc::GetModule().
I checked performance. It is slower than PreferredZapModule over 5%.

Reply:
Why it is slower?

Reply:
A generic method is encoded based on the module. So the module should know all types in a generic method.
For example, if List<TizenElement>.Add() List is in System module and TizenElement is a struct, MethodDesc::GetModule will return System module. System module doesn't know TizenElement, so it fails to encode. However, PreferredZapModule may return a module of TizenElement.

Reply:
> So the module should know all types in a generic method.

This constraint is specific to ngen signature encoding. Signature encoding for multicore jit trace should not need to be limited by it.
.

Reply:
> This constraint is specific to ngen signature encoding. 

Yes, I used ZapSig::EncodeMethod to encode the generic method. It has the constraint.
> Signature encoding for multicore jit trace should not need to be limited by it.

You mean it needs new signature encoding for multicore jit using exising signature builder? Or is there any other encoding function which it can use?

Reply:
I think you can reuse most of the encoding logic from `ZapSig::EncodeMethod`, but without depending on the preferred zap module algorithm.

Reply:
Thank you. I think that I can make encoding logic that handles value type instantiation (if cannot access from the module of the generic method) in a generic method differently.